components/browser/menu/src/main/java/mozilla/components/browser/menu/item/BrowserMenuItemToolbar.kt

```diff
@@ -30,27 +30,42 @@ class BrowserMenuItemToolbar(
 
         for (item in items) {
             val button = AppCompatImageButton(view.context)
-            button.setImageResource(item.imageResource)
+            if (item is TwoStateButton && !item.isInPrimaryState.invoke()) {
```

Comment:
This can get complex really quickly if we do type and state checks. Maybe we can create a `BrowserMenuTwoStateButton`? We have an abstract `BrowserMenuCompoundButton` that might be useful. It's currently only used for `BrowserMenuCheckbox` and `BrowserMenuSwitch`.

We also have `BrowserToolbar.TwoStateButton` that might have an API similar to one that you need that can be copied for the BrowserMenu.